Mage.Tests/src/test/java/org/mage/test/cards/copy/CopySpellTest.java

```diff
@@ -77,6 +77,29 @@ public void ZadaHedronGrinderBoost() {
         assertAbility(playerB, "Silvercoat Lion", FlyingAbility.getInstance(), false);
     }
 
+    @Test
+    public void BonecrusherGiantChangeTargets() {
+        addCard(Zone.BATTLEFIELD, playerA, "Bonecrusher Giant");
+        addCard(Zone.BATTLEFIELD, playerA, "Grizzly Bears");
+        addCard(Zone.BATTLEFIELD, playerA, "Savannah Lions");
+        addCard(Zone.BATTLEFIELD, playerA, "Plains");
+        addCard(Zone.HAND, playerA, "Barkshell Blessing");
+
+        castSpell(1, PhaseStep.UPKEEP, playerA, "Barkshell Blessing");
+        setChoice(playerA, "Yes");
+        setChoice(playerA, "Yes");
+        addTarget(playerA, "Grizzly Bears");
+        addTarget(playerA, "Bonecrusher Giant");
+
+        setStopAt(1, PhaseStep.PRECOMBAT_MAIN);
+        execute();
```

Comment:
Use `setStrictChooseMode` and `assertAllCommandsUsed` all the time in tests (only one exception: AI tests):
```
        setStrictChooseMode(true);
        setStopAt(1, PhaseStep.END_COMBAT);
        execute();
        assertAllCommandsUsed();
```